src/cookieSyncManager.ts

```diff
@@ -0,0 +1,184 @@
+import { isEmpty, replaceAmpWithAmpersand, replaceMPID } from './utils';
+import Constants from './constants';
+import { ICookieSyncManager } from './cookieSyncManager.interfaces';
+import { MParticleWebSDK } from './sdkRuntimeModels';
+import { Dictionary, MPID } from '@mparticle/web-sdk';
+import { IConsentRules } from './consent';
+
+const { Messages } = Constants;
+const { InformationMessages } = Messages;
+
+interface PixelConfig {
+    moduleId: string;
+    frequencyCap: number;
+    pixelUrl: string;
+    redirectUrl: string;
+    filteringConsentRuleValues: IConsentRules;
+}
+
+export const DAYS_IN_MILLISECONDS = 1000 * 60 * 60 * 24;
+
+const hasFrequencyCapExpired = (
+    lastSyncDate: number,
+    frequencyCap: number
+): boolean => {
+    return (
+        new Date().getTime() >
+        new Date(lastSyncDate).getTime() + frequencyCap * DAYS_IN_MILLISECONDS
+    );
```

Comment:
`lastSyncDate` should be optional, and the second parameter.  [this code block](https://github.com/mParticle/mparticle-web-sdk/pull/944/commits/b6a0be6bfebb5ffce2dad7cd853dbc2e89cf472d#diff-240c8f092446a9912eb4e9046db7089f14af869309570f62a42cd6e9f0e75c90L108-L143) will call `self.performCookieSync` if there is no `lastSyncDate` and